# Sales Correlation Analysis for WaterPro Company

**Background and Task Overview**

WaterPro, a company that introduced two new products, WaterCure and BobaWonder, to the market in January 2024, is interested in understanding the relationship between the sales of these two products. They want to know if good sales performance for BobaWonder correlates with good sales for WaterCure. In other words, the company is curious to see if there is a pattern where high sales of one product are associated with high sales of the other.

**Motivation for Correlation Analysis**

Correlation analysis allows us to assess the strength and direction of the relationship between two variables — in this case, the monthly sales revenues of BobaWonder and WaterCure. By calculating the Pearson correlation coefficient, we can determine whether:

- The sales of BobaWonder and WaterCure move together in a positive direction (positive correlation),
- They move in opposite directions (negative correlation),Or if they are largely unrelated (no significant correlation).
- A strong correlation could imply that similar factors influence the sales of both products, whereas a weak or negative correlation may suggest that -their sales are driven by different factors.

**Objectives**
 To provide insights for WaterPro, we will:

- Create dictionaries to store the monthly sales data for BobaWonder and WaterCure.
- Identify the months with the highest and lowest sales for both products.
- Find the months where sales exceeded 750 units for each product.
- Calculate the Pearson correlation coefficient to determine if there is a relationship between the sales of BobaWonder and WaterCure.
- Interpret the results based on the correlation coefficient to offer meaningful insights to the company.

In [4]:
# Import necessary libraries
import scipy.stats 


boba_sales = {          # Step 1: Creating dictionaries for BobaWonder and WaterCure
    'January': 900, 
    'February': 800, 
    'March': 639, 
    'April': 772, 
    'May': 682, 
    'June': 635, 
    'July': 538, 
    'August': 716
}

watercure_sales = {
    'January': 890, 
    'February': 700, 
    'March': 523, 
    'April': 612, 
    'May': 661, 
    'June': 456, 
    'July': 435, 
    'August': 452
}

max_boba_sales_month = max(boba_sales, key=boba_sales.get)   # Step 2: Finding  months with the higest and lowest sales for both products

min_boba_sales_month = min(boba_sales, key=boba_sales.get)

max_watercure_sales_month = max(watercure_sales, key=watercure_sales.get)
min_watercure_sales_month = min(watercure_sales, key=watercure_sales.get)


print(f"Highest sales for BobaWonder: {max_boba_sales_month} ({boba_sales[max_boba_sales_month]})") #Step 3  Printing the result for highest and lowest sales
print(f"Lowest sales for BobaWonder: {min_boba_sales_month} ({boba_sales[min_boba_sales_month]})")

print(f"Highest sales for WaterCure: {max_watercure_sales_month} ({watercure_sales[max_watercure_sales_month]})")
print(f"Lowest sales for WaterCure: {min_watercure_sales_month} ({watercure_sales[min_watercure_sales_month]})")


print("\nMonths with BobaWonder sales over 750:")   # Step 4: Printing months where sales for each product are greater than 750
for month, sales in boba_sales.items():
    if sales > 750:
        print(f"{month}: {sales}")

print("\nMonths with WaterCure sales over 750:")
for month, sales in watercure_sales.items():
    if sales > 750:
        print(f"{month}: {sales}")



boba_sales_values = list(boba_sales.values()) # Step 5: Calculating Pearson correlation coefficient and extracting sales data as lists
watercure_sales_values = list(watercure_sales.values())


correlation_coefficient, p_value = pearsonr(boba_sales_values, watercure_sales_values)   #Step 6: Calculating Pearson correlation coefficient


print(f"\nPearson correlation coefficient: {correlation_coefficient}") #Step 7 Printing the correlation coefficient and p-value
print(f"P-value: {p_value}")


Highest sales for BobaWonder: January (900)
Lowest sales for BobaWonder: July (538)
Highest sales for WaterCure: January (890)
Lowest sales for WaterCure: July (435)

Months with BobaWonder sales over 750:
January: 900
February: 800
April: 772

Months with WaterCure sales over 750:
January: 890

Pearson correlation coefficient: 0.8607420030427412
P-value: 0.006065991649604092


***Interpretation of the Results:***

**Highest and Lowest Sales:**

- BobaWonder had its highest sales in January with 900 units and its lowest sales in July with 538 units.
- WaterCure had its highest sales in January with 890 units and its lowest sales in July with 435 units.
- Months with Sales Over 750:
- BobaWonder: The sales exceeded 750 units in January, February, and April.
- WaterCure: Only January recorded sales over 750 units.

**Pearson Correlation Coefficient:**

- The Pearson correlation coefficient is 0.8607 close to 1, which indicates a strong positive linear relationship between the sales of BobaWonder and WaterCure. This means that when the sales of BobaWonder increase, the sales of WaterCure tend to also increase, and vice versa.
- The p-value is very small (likely below 0.05, though not explicitly stated), suggesting that this correlation is statistically significant. A significant p-value means the strong correlation between the sales of BobaWonder and WaterCure is unlikely to have occurred by random chance.
  the other.


Thus, the sales performance of BobaWonder is positively correlated with that of WaterCure, and good sales of one product typically indicate good sales for the other.

**Summary**

This explanation provides a summary of the findings from the code analysis,highlighting the key insights regarding sales performance of BobaWonder and WaterCure and the implications for WaterPro's business strategy